src/parsers/forecast_parser.jl

```diff
@@ -293,3 +293,42 @@ function parse_json(filename,device_names)
     return Devices
 end
 =#
+
+"""
+    remove_all_forecasts(sys::System, T::Type{T})::Vector{T} where T <: Forecast
+
+Remove all forecast objects of concrete type T from a System
+"""
+function remove_all_forecasts!(sys::System, ForecastType::Type{T})where T <: Forecast
+
+    ts = get_forecast_initial_times(sys)
```

Comment:
Do we need to iterate to remove all the forecasts? Can't we just overwrite the data with an empty dict?

Reply:
Even better:  ```empty!(sys.forecasts.data)```

Reply:
I made this slightly more specific so that we can remove all forecasts of a particular type. If that's not a valid use case, I'll replace this with @daniel-thom's suggestion. Otherwise, I'll implement the suggestion in another method that just takes the `sys` argument.